sdk/schemaregistry/Microsoft.Azure.Data.SchemaRegistry.ApacheAvro/tests/SchemaRegistryAvroObjectSerializerLiveTests.cs

```diff
@@ -74,15 +72,69 @@ public async Task CannotDeserializeIntoNonCompatibleType()
         {
             var client = CreateClient();
             var groupName = TestEnvironment.SchemaRegistryGroup;
-            var employee = new Employee() { Age = 42, Name = "Caketown"};
+            var employee = new Employee() { Age = 42, Name = "Caketown" };
 
             var serializer = new SchemaRegistryAvroSerializer(client, groupName, new SchemaRegistryAvroSerializerOptions { AutoRegisterSchemas = true });
             var content = await serializer.SerializeAsync<BinaryContent, Employee>(employee);
+            var schemaId = content.ContentType.ToString().Split('+')[1];
 
             // deserialize with the new schema, which is NOT backward compatible with the old schema as it adds a new field
             Assert.That(
                 async () => await serializer.DeserializeAsync<Employee_V2>(content),
-                Throws.InstanceOf<AvroException>());
+                Throws.InstanceOf<AvroSerializationException>()
+                    .And.Property(nameof(Exception.InnerException)).InstanceOf<AvroException>()
+                    .And.Property(nameof(AvroSerializationException.SerializedSchemaId)).EqualTo(schemaId));
+
+            #region Snippet:SchemaRegistryAvroException
+            try
+            {
+                Employee_V2 employeeV2 = await serializer.DeserializeAsync<Employee_V2>(content);
+            }
+            catch (AvroSerializationException exception)
+            {
+                // When this exception occurs when deserializing, the exception message will contain the schema ID that was used to
+                // serialize the data.
+                Console.WriteLine(exception);
```

Comment:
should we assert on the exception message?

Reply:
Ah, I missed this. I didn't put an assertion here because this gets pulled into our snippets. I generally avoid asserting on the contents of the exception message anyway because it doesn't add a ton of value.